src/main/java/org/apache/commons/imaging/formats/png/PngImageParser.java

```diff
@@ -100,119 +161,97 @@ public String getDefaultExtension() {
-            final int chunkType = read4Bytes("ChunkType", is, "Not a Valid PNG File", getByteOrder());
-
-            if (LOGGER.isLoggable(Level.FINEST)) {
-                printCharQuad("ChunkType", chunkType);
-                debugNumber("Length", length, 4);
-            }
-            final boolean keep = keepChunk(chunkType, chunkTypes);
-
-            byte[] bytes = null;
-            if (keep) {
-                bytes = readBytes("Chunk Data", is, length,
-                        "Not a Valid PNG File: Couldn't read Chunk Data.");
-            } else {
-                skipBytes(is, length, "Not a Valid PNG File");
-            }
-
-            if (LOGGER.isLoggable(Level.FINEST)) {
-                if (bytes != null) {
-                    debugNumber("bytes", bytes.length, 4);
-                }
-            }
-
-            final int crc = read4Bytes("CRC", is, "Not a Valid PNG File", getByteOrder());
-
-            if (keep) {
-                if (chunkType == ChunkType.iCCP.value) {
-                    result.add(new PngChunkIccp(length, chunkType, crc, bytes));
-                } else if (chunkType == ChunkType.tEXt.value) {
-                    result.add(new PngChunkText(length, chunkType, crc, bytes));
-                } else if (chunkType == ChunkType.zTXt.value) {
-                    result.add(new PngChunkZtxt(length, chunkType, crc, bytes));
-                } else if (chunkType == ChunkType.IHDR.value) {
-                    result.add(new PngChunkIhdr(length, chunkType, crc, bytes));
-                } else if (chunkType == ChunkType.PLTE.value) {
-                    result.add(new PngChunkPlte(length, chunkType, crc, bytes));
-                } else if (chunkType == ChunkType.pHYs.value) {
-                    result.add(new PngChunkPhys(length, chunkType, crc, bytes));
-                } else if (chunkType == ChunkType.sCAL.value) {
-                    result.add(new PngChunkScal(length, chunkType, crc, bytes));
-                } else if (chunkType == ChunkType.IDAT.value) {
-                    result.add(new PngChunkIdat(length, chunkType, crc, bytes));
-                } else if (chunkType == ChunkType.gAMA.value) {
-                    result.add(new PngChunkGama(length, chunkType, crc, bytes));
-                } else if (chunkType == ChunkType.iTXt.value) {
-                    result.add(new PngChunkItxt(length, chunkType, crc, bytes));
-                } else {
-                    result.add(new PngChunk(length, chunkType, crc, bytes));
+        final DataInputStream dataInput = new DataInputStream(inputStream);
+        
+        while(true) {
+            final ChunkLoader nextChunkLoaded = ChunkLoader.run(dataInput, selector);
+            PngChunk loadedChunk = nextChunkLoaded.loadedChunk();
+            
+            if(loadedChunk != null) {
+                { // (Shukant Pal): I essentially hate how developers
```

Comment:
Author name not needed. And see note about this kind of note in a previous comment. We must be polite and nice to each developer, no matter how good or bad their code can be.